sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableCatalog.java

```diff
@@ -222,6 +226,7 @@ default Table createTable(
    * @throws UnsupportedOperationException If a requested partition transform is not supported
    * @throws NoSuchNamespaceException If the identifier namespace does not exist (optional)
    */
+  @Deprecated(since = "4.0.0")
```

Comment:
This is a backward compatible API refactor, shall we just put it in 4.1.0?

Reply:
SG

Reply:
I agree, it will be shipped in 4.1, so it can't be deprecated since 4.0.